http-client/src/client.rs

```diff
@@ -75,7 +75,7 @@ impl HttpClientBuilder {
 			.map_err(|e| Error::Transport(e.into()))?;
 		Ok(HttpClient {
 			transport,
-			id_manager: RequestIdManager::new(self.max_concurrent_requests),
+			id_manager: Arc::new(RequestIdManager::new(self.max_concurrent_requests)),
```

Comment:
dq: could this be a `Box` too?

Reply:
Maybe with:

```
impl Clone for RequestIdManager {
	fn clone(&self) -> Self {
		RequestIdManager {
			current_pending: self.current_pending.clone(),
			max_concurrent_requests: self.max_concurrent_requests.clone(),
			current_id: AtomicU64::new(self.current_id.load(std::sync::atomic::Ordering::SeqCst)),
		}
	}
}
```

Reply:
Because we rely on the strong count on `current_pending` for how many concurrent requests cloning `current_pending` it increase the `strong count` and it will not work properly.

Thus, that's why I used  `Arc<RequestIdManager>`.

Why is `Box` better?

Reply:
> Box only works when RequestIdManager is clone right?! I don't want it to be Clone because the reason ☝️

Ah, right, ok, I get it now. 

Yes, `RequestIdManager` would have to be `Clone` (and if it were you wouldn't need nor `Box` or `Arc`).